src/pkcs15init/pkcs15-openpgp.c

```diff
@@ -349,7 +338,8 @@ static int openpgp_generate_key_ec(sc_card_t *card, sc_pkcs15_object_t *obj,
 	pubkey->u.ec.ecpointQ.len = key_info.u.ec.ecpoint_len;
 	pubkey->u.ec.ecpointQ.value = malloc(key_info.u.ec.ecpoint_len);
 	if (pubkey->u.ec.ecpointQ.value == NULL)
-		goto err;
+		LOG_FUNC_RETURN(ctx, SC_ERROR_NOT_ENOUGH_MEMORY);
```

Comment:
You are not cleaning up as before.
Should this not be
r = SC_ERROR_NOT_ENOUGH_MEMORY; goto err;

Reply:
Yes  will correct.

Reply:
Fixed.